api/src/tests/conftest.py

```diff
@@ -24,32 +27,50 @@ def test_app():
 
 
 def pytest_addoption(parser):
+    """
+    Adds option flags to pytest:
+        --integration: runs only integration tests
+        --unit: runs unit tests
+    """
     parser.addoption("--integration", action="store_true", help="run integration tests")
+    parser.addoption("--unit", action="store_true", help="run unit tests")
 
 
 def pytest_collection_modifyitems(config, items):
-    if config.getoption("--integration"):
-        # --integration given in cli
-        for item in items:
-            if "integration" not in item.keywords:
+    """
+    Collection modifier for passed options. If --integration is passed, only integration tests are run, similarly
+    for --unit. No options mean all tests are run.
+    """
+    for item in items:
+        if config.getoption("--integration"):
+            if "integration" not in str(item.path):
                 item.add_marker(
-                    pytest.mark.skip(reason="--integration option is passed: Test is not an integration test")
+                    pytest.mark.skip(reason=f"--integration option is passed: {item.name} is not an integration test")
                 )
-    else:
-        for item in items:
-            if "integration" in item.keywords:
-                item.add_marker(pytest.mark.skip(reason="need --integration option to run"))
+        elif config.getoption("--unit"):
+            if "unit" not in str(item.path):
+                item.add_marker(pytest.mark.skip(reason=f"--unit option is passed: {item.name} is not a unit test"))
 
 
 def read_json(filepath: str):
+    """Returns the contents of a json file at the filepath given"""
     with open(filepath, "r") as file:
         data = file.read()
     return json.loads(data)
 
 
 @pytest.fixture(scope="module")
-def get_multiflash_test_data():
+def get_multiflash_test_data(request) -> (Dict, MultiflashResult):
```

Comment:
And this could be renamed to "convert_response_data_to_classes", or something....

Reply:
Went for something along those lines. The only "iffy" part is that the data from the output which is converted is not strictly the same as the response data (the response data does not contain mass concentrations). Not too sure if this is a minuscule non-issue or if the function should be renamed.

Reply:
Right. IMO the test data should not require any conversion. It should be the exact API response.
If we need additional test data for testing the library output, that should be a separate test with separate data.
Any good reasons why this is not the case?

Reply:
Reason (not exactly good): trying to be elegant.